ocaml/xapi/test_clustering.ml

```diff
@@ -207,11 +207,60 @@ let test_assert_cluster_host_is_enabled_for_matching_sms =
   ; "test_assert_cluster_host_is_enabled_for_matching_sms_succeeds_if_no_cluster_host_exists_and_clustering_is_not_needed" >:: test_assert_cluster_host_is_enabled_for_matching_sms_succeeds_if_no_cluster_host_exists_and_clustering_is_not_needed
   ]
 
+
+(** Tests clustering lock is only taken if needed *)
+let test_clustering_lock_only_taken_if_needed = 
+  let nest_with_clustering_lock_if_needed ~__context ~type1 ~type2 ~on_deadlock ~on_no_deadlock =
+    Helpers.timebox
+      ~timeout: 1.0
```

Comment:
Lets make this one smaller (e.g. `0.1`) if we do intend to detect the deadlocks during regular unit-testing